tests/test_dataset_report.py

```diff
@@ -35,24 +35,38 @@ def test_get_name(path, name):
     assert dataset_report.get_name(path) == name
 
 
-class TestIsBool:
-    def test_with_boolean(self):
-        assert dataset_report.is_boolean(pandas.Series([0, 1], dtype=int))
-        assert dataset_report.is_boolean(pandas.Series([0, 1], dtype=float))
-        assert dataset_report.is_boolean(pandas.Series([numpy.nan, 1], dtype=float))
-        assert dataset_report.is_boolean(pandas.Series([False, True], dtype=bool))
-
-    def test_with_non_boolean(self):
-        assert not dataset_report.is_boolean(pandas.Series([0, 2], dtype=int))
-        assert not dataset_report.is_boolean(pandas.Series([0.1, 0.2], dtype=float))
-        assert not dataset_report.is_boolean(pandas.Series([numpy.nan, 2], dtype=float))
-        assert not dataset_report.is_boolean(pandas.Series(["0", "1"], dtype=str))
-        assert not dataset_report.is_boolean(
-            pandas.Series(
-                [
-                    datetime.datetime(2022, 1, 1),
-                    datetime.datetime(2022, 1, 2),
-                ],
-                dtype="datetime64[ns]",
-            )
-        )
+class TestIsBoolean:
+    @pytest.mark.parametrize(
+        "data,dtype",
+        [
+            ([0, 1], int),
+            ([0, 1], float),
+            ([numpy.nan, 1], float),
+            ([False, True], bool),
+            # We have no way of knowing whether the following series should contain
+            # boolean values when it only contains missing values. However, the
+            # distinction doesn't matter in practice.
+            ([numpy.nan, numpy.nan], float),
+        ],
+    )
+    def test_with_boolean_values(self, data, dtype):
+        assert dataset_report.is_boolean(pandas.Series(data, dtype=dtype))
+
+    @pytest.mark.parametrize(
+        "data,dtype",
+        [
+            ([0, 2], int),
+            ([0.1, 0.2], float),
+            ([numpy.nan, 2], float),
+            (["0", "1"], str),
```

Comment:
Am I understanding correctly that we _won't_ be consider ["0", "1"] as bool? Should we want to? I am trying to think of a case where we have "0" and "1" as the unique values in a series, and it doesn't serve as an indicator/effectively a bool.

Reply:
That's correct, and I don't think we want to 🙂. We know that a boolean series can be represented as a numeric series, with values drawn from the set `{0, 1, numpy.nan}`. A case where a boolean series is represented as a string (`object`) series would be unusual. I'd prefer us not to make any assumptions about one until we see one.

Reply:
Ultimately, we'd probably want to perform the same operations on a numeric-as-boolean as a string-as-boolean. Keeping the two separated, however, hopefully emphasizes that the former is usual, but the latter is unusual.

Reply:
Got it, that makes sense. Then I am happy to approve!